modules/light-clients/xx-mock/types/client_state.go

```diff
@@ -113,7 +115,38 @@ func (cs ClientState) VerifyMembership(
 		return sdkerrors.Wrap(clienttypes.ErrConsensusStateNotFound, "please ensure the proof was constructed against a height that exists on the client")
 	}
 
-	h := sha256.Sum256(value)
+	// sha256(abi.encodePacked(height.toUint128(), sha256(prefix), sha256(path), sha256(value)))
+	revisionNumber := height.GetRevisionNumber()
+	revisionHeight := height.GetRevisionHeight()
+
+	heightBig := new(big.Int)
+	revisionNumberBig := new(big.Int).SetUint64(revisionNumber)
+	revisionNumberBig = revisionNumberBig.Lsh(revisionNumberBig, 64)
+	revisionHeightBig := new(big.Int).SetUint64(revisionHeight)
+	heightBig.Or(revisionNumberBig, revisionHeightBig)
+	hashHeight := sha256.Sum256(heightBig.Bytes())
```

Comment:
There are two mistakes.
- Look at the spec again. At first `height` is concatenated to the other stuffs before hashing.
- The byte length of `heightBig.Bytes()` can be less than 16.
    - For example, if `revisionNumber == 0`, it will be 8.
    - You should use `FillBytes` instead of `Bytes`.

Reply:
I think you can use `encoding/binary` to do this easilier.
https://pkg.go.dev/encoding/binary#example-ByteOrder-Put

Reply:
Thanks
Change to use encoding/binary
https://github.com/datachainlab/ibc-mock-client/pull/11/commits/28f2d67bd4974b3582fcd68a70456281873a7d4d